libvast/src/system/active_partition.cpp

```diff
@@ -412,114 +539,12 @@ active_partition_actor::behavior_type active_partition(
                            *self, self->state.persisted_indexers, valid_count);
                 return;
```

Comment:
The old code was not pretty here, nowadays we do have `detail::fanout_counter` to handle this pattern. It may make sense to use it here and then call `serialize()` from its completion handler.